src/languages/gml.js

```diff
@@ -2994,135 +937,572 @@ export default function(hljs) {
-    "phy_collision_x",
-    "phy_collision_y",
+    "phy_fixed_rotation",
+    "phy_active",
+    "phy_mass",
+    "phy_inertia",
     "phy_com_x",
     "phy_com_y",
     "phy_dynamic",
-    "phy_fixed_rotation",
-    "phy_inertia",
     "phy_kinematic",
-    "phy_linear_damping",
-    "phy_linear_velocity_x",
-    "phy_linear_velocity_y",
-    "phy_mass",
-    "phy_position_x",
+    "phy_sleeping",
+    "phy_collision_points",
+    "phy_collision_x",
+    "phy_collision_y",
+    "phy_col_normal_x",
+    "phy_col_normal_y",
     "phy_position_xprevious",
-    "phy_position_y",
     "phy_position_yprevious",
-    "phy_rotation",
-    "phy_sleeping",
-    "phy_speed",
-    "phy_speed_x",
-    "phy_speed_y",
-    "player_avatar_sprite",
-    "player_avatar_url",
     "player_id",
     "player_local",
-    "player_type",
+    "player_avatar_url",
+    "player_avatar_sprite",
     "player_user_id",
-    "program_directory",
-    "rollback_api_server",
-    "rollback_confirmed_frame",
+    "player_type",
+    "managed",
     "rollback_current_frame",
     "rollback_event_id",
     "rollback_event_param",
     "rollback_game_running",
-    "room",
-    "room_first",
-    "room_height",
-    "room_last",
-    "room_persistent",
-    "room_speed",
-    "room_width",
-    "score",
-    "script",
-    "sequence_instance",
-    "solid",
-    "speed",
-    "sprite_height",
-    "sprite_index",
-    "sprite_width",
-    "sprite_xoffset",
-    "sprite_yoffset",
-    "stacktrace",
-    "temp_directory",
-    "timeline_index",
-    "timeline_loop",
-    "timeline_position",
-    "timeline_running",
-    "timeline_speed",
-    "view_camera",
-    "view_current",
-    "view_enabled",
-    "view_hport",
-    "view_surface_id",
-    "view_visible",
-    "view_wport",
-    "view_xport",
-    "view_yport",
-    "visible",
-    "vspeed",
-    "webgl_enabled",
-    "working_directory",
-    "x",
-    "xprevious",
-    "xstart",
-    "y",
-    "yprevious",
-    "ystart"
+    "rollback_confirmed_frame",
+    "rollback_api_server",
+    "player_avatar_sprite",
+    "player_avatar_url",
+    "player_user_id",
+    "player_type"
+  ];
+  /**
+   * Regex for some sort of identifier - i.e, a valid name of something in code.
+   */
+  const VALID_IDENTIFIER_REG = /[a-zA-Z_][a-zA-Z0-9_]*/;
+  /**
+   * Regex for a dot separating some LHS and RHS expression with optional whitespace (as this is
+   * supported in the engine.)
+   */
+  const DOT_ACCESSOR_REG = /\b\.\b/;
+
+  /**
+   * Expressions, which form part of a valid statement.
+   * @type {Object[]}
+   */
+  const EXPRESSION = [];
+
+  /**
+   * A single-line comment.
+   */
+  const COMMENT_LINE = hljs.COMMENT('//', /\$|\n/);
+
+  const JSDOC_TYPE_PARAM_REG = /{ *?[a-zA-Z_][a-zA-Z0-9_\.<>\|]*? *?} *?/;
+
+  const JSDOC_ANNOTATIONS = [
+    {
+      match: /@((desc(ription)?)|ignore|pure|deprecated|(func(tion)?))\b/,
+      className: "keyword"
+    },
+    {
+      match: [
+        /@(self|context) *?/,
```

Comment:
What is the goal of ` *?`?

Reply:
I'm matching 0 or more spaces there, so that any of these are valid:
```gml
/// @self {Struct.Thing}
/// @self                                  {Struct.Thing}
/// @self{Struct.Thing}
```
...which matches the behaviour of GameMaker's IDE. I tried sticking `\b` there instead, but the exact behaviour of `\b` continues to elude me, so as it didn't work, I just replaced it with that instead.

Reply:
Right, but why ` *?` vs just ` *`... Isn't the default of greedy just fine here without the `?`?

Reply:
To be honest I dunno why I wrote that. I think I got in my head at some point or other that `?` was "generally preferable" for no particular reason. In this case it does absolutely nothing, so I'll take that out!